lib/modules/versioning/poetry/transform.ts

```diff
@@ -63,12 +63,18 @@ export function poetry2semver(
 
   const parts = [releaseParts.map((num) => num.toString()).join('.')];
   if (pre !== null) {
+    // trim leading zeros from valid numbers
+    pre.number = parseInt(pre.number, 10).toString();
```

Comment:
are you sure there's always a valid number?

what about `1.2.3pre`

Reply:
I will add a test for this but I think https://github.com/renovatebot/renovate/blob/132db01f14ce5c30609b8e82279ab14c9da9e8ff/lib/modules/versioning/poetry/transform.ts#L24 will guarantee it is at least a zero once parsed.

Reply:
I added a couple more tests but I think this is okay